commit 51f52a47527a07e45746b1bac1ab6123892095a0
Author: Anton Blanchard <anton@samba.org>
Date:   Mon May 9 10:07:40 2011 +1000

    [SCSI] ipr: Rate limit DMA mapping errors
    
    I noticed a stream of errors from the IPR driver while doing
    IOMMU fault injection. Rate limit the errors so we don't clog
    up the console and logfiles.
    
    Signed-off-by: Anton Blanchard <anton@samba.org>
    Acked-by: Brian King <brking@linux.vnet.ibm.com>
    Signed-off-by: James Bottomley <jbottomley@parallels.com>
    Signed-off-by: James Bottomley <James.Bottomley@suse.de>

diff --git a/drivers/scsi/ipr.c b/drivers/scsi/ipr.c
index 3667f89abdea..12868ca46110 100644
--- a/drivers/scsi/ipr.c
+++ b/drivers/scsi/ipr.c
@@ -5200,7 +5200,8 @@ static int ipr_build_ioadl64(struct ipr_ioa_cfg *ioa_cfg,
 
 	nseg = scsi_dma_map(scsi_cmd);
 	if (nseg < 0) {
-		dev_err(&ioa_cfg->pdev->dev, "pci_map_sg failed!\n");
+		if (printk_ratelimit())
+			dev_err(&ioa_cfg->pdev->dev, "pci_map_sg failed!\n");
 		return -1;
 	}